intro-to-oop-contd/projects-with-many-files.md

```diff
@@ -147,40 +171,176 @@ We must _import_ any resources that we need from another module before we can us
+
+<br />
+
+Hopefully good naming practices were employed to help hint at the proper usage of different identifiers. But if we import an identifier that refers to a value (a variable) but then try to call it like a function, this will result in a runtime error! So it's our job to read and understand the code or documentation of the module we want to use so that we import and make use of its identifiers correctly!
 
 ### !end-callout
 
-### Example Between Classes
+#### 3. Importing Module Identifiers using Relative Names
+
+Importing package modules by full name works everywhere, whether from outside or inside a package. But to access the identifiers defined in the module, we have to prefix them with the full package name. If our package were several layers deep, this could make for some long names!
+
+Let's say that the `passenger` module in our `ride_share_app` package needs to make use of the `Driver` class. We could certainly import the `driver` module as before and access the `Driver` class with the full package name. But `driver.py` and `passenger.py` are defined in the same package. In fact, they're in the same folder. Wouldn't it be nice if there were a shorter way?
+
+There is!
+
+```python
+from .driver import Driver
+```
+
+This may look a little weird. It might even seem backwards! Feel free to read this statement from right to left if that's helpful!
+
+What this statement does is:
+
+1. Use the `from` keyword to name the module from which we want to import an identifier.
+1. Use the `import` keyword to name the identifier we want Python to import into the current scope
+
+The module name looks different from the examples we've seen before. It starts with a `.` which tells Python to look for the module in the folder containing the current file. When Python finds the specified module, rather than importing the name of the module into the current scope, it looks at the names listed after `import` (we can write multiple names separated by commas) and imports just those identifiers into the current scope.
+
+If we assume as before that the `Driver` identifier is a class, this lets us write the following:
+
+```python
+driver = Driver()
+```
+
+Shorter and clearer!
+
+```
+└── ride_share_app
+    ├── __init__.py
+    ├── driver.py
+    ├── passenger.py
+    └── payment
+        ├── __init__.py
+        ├── cash.py
+        └── credit.py
+```
+
+If our package had a folder called `payment` under `ride_share_app` (with an `__init__.py` file), and `payment` had the modules `cash.py` and `credit.py`, then from either of these modules we could access the `Driver` identifier in `driver.py` as:
```

Comment:
My preference is to think of this `payment` section as a second paragraph.

With that, I think I'd recommend:

- Moving this line to directly above the example tree structure (currently directly above this paragraph)
- Modifying the wording of this line to introduce this as a second example
- Frame this second example to solving a different question: What's the syntax for importing something from two folders up?
- Name the modules `cash.py` and `credit.py` in a simplified sense instead of saying "then from either of these modules." (take a more concrete stance, so a student brain doesn't have to parse through "is it for cash.py or credit.py, what is a module, which folder are we looking at," etc)

```suggestion
Let's consider a second, similar situation. How do we import a module that is inside a different folder, but within the same package?

Imagine that our package had a folder called `payment` under `ride_share_app` (with an `__init__.py` file), and `payment` had the modules `cash.py` and `credit.py`. Inside either `cash.py` or `credit.py`, we can access the `Driver` identifier in `driver.py` as:
```

Reply:
Taking the suggestion. I'll move the lines around locally.